### ISSUE TITLE
lon dimension error when applying combined_preprocessing for some CMIP6 models 

### ISSUE THREAD

**Issue by dgergel** [jbusecke/cmip6_preprocessing#98]
When I use `combined_preprocessing` in conjunction with CMIP6 model output in the example below: 

```
import xarray as xr
import numpy as np 
import os 

import intake
import zarr
from cmip6_preprocessing.preprocessing import combined_preprocessing

# search the catalog and get model data
col = intake.open_esm_datastore("https://storage.googleapis.com/cmip6/pangeo-cmip6.json")
cat = col.search(activity_id='ScenarioMIP', institution_id='EC-Earth-Consortium', source_id='EC-Earth3-Veg-LR', experiment_id='ssp370', table_id='day', variable_id='tasmax', 
                 member_id='r1i1p1f1')

z_kwargs = {'consolidated': True, 'decode_times':False}
dset_dict = cat.to_dataset_dict(zarr_kwargs=z_kwargs,
                                preprocess=combined_preprocessing)
```

I'm getting the following `MissingDimensionsError`: 

```
MissingDimensionsError                    Traceback (most recent call last)
/srv/conda/envs/notebook/lib/python3.8/site-packages/intake_esm/merge_util.py in _open_asset()
    314         try:
--> 315             ds = preprocess(ds)
    316         except Exception as exc:

/srv/conda/envs/notebook/lib/python3.8/site-packages/cmip6_preprocessing/preprocessing.py in combined_preprocessing()
    904         # broadcast lon/lat
--> 905         ds = broadcast_lonlat(ds)
    906         # replace x,y with nominal lon,lat

/srv/conda/envs/notebook/lib/python3.8/site-packages/cmip6_preprocessing/preprocessing.py in broadcast_lonlat()
    814     if len(ds["lon"].dims) < 2:
--> 815         ds.coords["lon"] = ds["lon"] * xr.ones_like(ds["lat"])
    816     if len(ds["lat"].dims) < 2:

/srv/conda/envs/notebook/lib/python3.8/site-packages/xarray/core/coordinates.py in __setitem__()
     39     def __setitem__(self, key: Hashable, value: Any) -> None:
---> 40         self.update({key: value})
     41 

/srv/conda/envs/notebook/lib/python3.8/site-packages/xarray/core/coordinates.py in update()
    114         other_vars = getattr(other, "variables", other)
--> 115         coords, indexes = merge_coords(
    116             [self.variables, other_vars], priority_arg=1, indexes=self.indexes

/srv/conda/envs/notebook/lib/python3.8/site-packages/xarray/core/merge.py in merge_coords()
    454     )
--> 455     collected = collect_variables_and_indexes(aligned)
    456     prioritized = _get_priority_vars_and_indexes(aligned, priority_arg, compat=compat)

/srv/conda/envs/notebook/lib/python3.8/site-packages/xarray/core/merge.py in collect_variables_and_indexes()
    277 
--> 278             variable = as_variable(variable, name=name)
    279             if variable.dims == (name,):

/srv/conda/envs/notebook/lib/python3.8/site-packages/xarray/core/variable.py in as_variable()
    153         if obj.ndim != 1:
--> 154             raise MissingDimensionsError(
    155                 "%r has more than 1-dimension and the same name as one of its "

MissingDimensionsError: 'lon' has more than 1-dimension and the same name as one of its dimensions ('lon', 'lat'). xarray disallows such variables because they conflict with the coordinates used to label dimensions.

The above exception was the direct cause of the following exception:

RuntimeError                              Traceback (most recent call last)
<ipython-input-30-3884fdb04db1> in <module>
      2 
      3 z_kwargs = {'consolidated': True, 'decode_times':False}
----> 4 dset_dict = cat.to_dataset_dict(zarr_kwargs=z_kwargs,
      5                                 preprocess=combined_preprocessing)
      6 # dset_dict = cat.to_dataset_dict(zarr_kwargs=z_kwargs)

/srv/conda/envs/notebook/lib/python3.8/site-packages/intake_esm/core.py in to_dataset_dict(self, zarr_kwargs, cdf_kwargs, preprocess, storage_options, progressbar, aggregate)
    925             ]
    926             for i, task in enumerate(concurrent.futures.as_completed(future_tasks)):
--> 927                 key, ds = task.result()
    928                 self._datasets[key] = ds
    929                 if self.progressbar:

/srv/conda/envs/notebook/lib/python3.8/concurrent/futures/_base.py in result(self, timeout)
    430                 raise CancelledError()
    431             elif self._state == FINISHED:
--> 432                 return self.__get_result()
    433 
    434             self._condition.wait(timeout)

/srv/conda/envs/notebook/lib/python3.8/concurrent/futures/_base.py in __get_result(self)
    386     def __get_result(self):
    387         if self._exception:
--> 388             raise self._exception
    389         else:
    390             return self._result

/srv/conda/envs/notebook/lib/python3.8/concurrent/futures/thread.py in run(self)
     55 
     56         try:
---> 57             result = self.fn(*self.args, **self.kwargs)
     58         except BaseException as exc:
     59             self.future.set_exception(exc)

/srv/conda/envs/notebook/lib/python3.8/site-packages/intake_esm/core.py in _load_source(key, source)
    911 
    912         def _load_source(key, source):
--> 913             return key, source.to_dask()
    914 
    915         sources = {key: source(**source_kwargs) for key, source in self.items()}

/srv/conda/envs/notebook/lib/python3.8/site-packages/intake_esm/source.py in to_dask(self)
    244     def to_dask(self):
    245         """Return xarray object (which will have chunks)"""
--> 246         self._load_metadata()
    247         return self._ds
    248 

/srv/conda/envs/notebook/lib/python3.8/site-packages/intake/source/base.py in _load_metadata(self)
    124         """load metadata only if needed"""
    125         if self._schema is None:
--> 126             self._schema = self._get_schema()
    127             self.datashape = self._schema.datashape
    128             self.dtype = self._schema.dtype

/srv/conda/envs/notebook/lib/python3.8/site-packages/intake_esm/source.py in _get_schema(self)
    173 
    174         if self._ds is None:
--> 175             self._open_dataset()
    176 
    177             metadata = {

/srv/conda/envs/notebook/lib/python3.8/site-packages/intake_esm/source.py in _open_dataset(self)
    225             for _, row in self.df.iterrows()
    226         ]
--> 227         datasets = dask.compute(*datasets)
    228         mapper_dict = dict(datasets)
    229         nd = create_nested_dict(self.df, self.path_column, self.aggregation_columns)

/srv/conda/envs/notebook/lib/python3.8/site-packages/dask/base.py in compute(*args, **kwargs)
    450         postcomputes.append(x.__dask_postcompute__())
    451 
--> 452     results = schedule(dsk, keys, **kwargs)
    453     return repack([f(r, *a) for r, (f, a) in zip(results, postcomputes)])
    454 

/srv/conda/envs/notebook/lib/python3.8/site-packages/distributed/client.py in get(self, dsk, keys, restrictions, loose_restrictions, resources, sync, asynchronous, direct, retries, priority, fifo_timeout, actors, **kwargs)
   2723                     should_rejoin = False
   2724             try:
-> 2725                 results = self.gather(packed, asynchronous=asynchronous, direct=direct)
   2726             finally:
   2727                 for f in futures.values():

/srv/conda/envs/notebook/lib/python3.8/site-packages/distributed/client.py in gather(self, futures, errors, direct, asynchronous)
   1984             else:
   1985                 local_worker = None
-> 1986             return self.sync(
   1987                 self._gather,
   1988                 futures,

/srv/conda/envs/notebook/lib/python3.8/site-packages/distributed/client.py in sync(self, func, asynchronous, callback_timeout, *args, **kwargs)
    830             return future
    831         else:
--> 832             return sync(
    833                 self.loop, func, *args, callback_timeout=callback_timeout, **kwargs
    834             )

/srv/conda/envs/notebook/lib/python3.8/site-packages/distributed/utils.py in sync(loop, func, callback_timeout, *args, **kwargs)
    338     if error[0]:
    339         typ, exc, tb = error[0]
--> 340         raise exc.with_traceback(tb)
    341     else:
    342         return result[0]

/srv/conda/envs/notebook/lib/python3.8/site-packages/distributed/utils.py in f()
    322             if callback_timeout is not None:
    323                 future = asyncio.wait_for(future, callback_timeout)
--> 324             result[0] = yield future
    325         except Exception as exc:
    326             error[0] = sys.exc_info()

/srv/conda/envs/notebook/lib/python3.8/site-packages/tornado/gen.py in run(self)
    760 
    761                     try:
--> 762                         value = future.result()
    763                     except Exception:
    764                         exc_info = sys.exc_info()

/srv/conda/envs/notebook/lib/python3.8/site-packages/distributed/client.py in _gather(self, futures, errors, direct, local_worker)
   1849                             exc = CancelledError(key)
   1850                         else:
-> 1851                             raise exception.with_traceback(traceback)
   1852                         raise exc
   1853                     if errors == "skip":

/srv/conda/envs/notebook/lib/python3.8/site-packages/intake_esm/source.py in read_dataset()
    202             # replace path column with mapper (dependent on filesystem type)
    203             mapper = _path_to_mapper(path, storage_options, data_format)
--> 204             ds = _open_asset(
    205                 mapper,
    206                 data_format=data_format,

/srv/conda/envs/notebook/lib/python3.8/site-packages/intake_esm/merge_util.py in _open_asset()
    315             ds = preprocess(ds)
    316         except Exception as exc:
--> 317             raise RuntimeError(
    318                 f'Failed to apply pre-processing function: {preprocess.__name__}'
    319             ) from exc

RuntimeError: Failed to apply pre-processing function: combined_preprocessing
```

When updating the `source_id` to `source_id='EC-Earth3'`, the error does not occur - it appears to be due to `combined_preprocessing` not being able to handle 2-d longitudes and the longitude dimension having the same name as the variable, even though `to_dataset_dict` can.

**Comment by jbusecke** [comment]
Dear @dgergel, thank you for using cmip6_preprocessing and raising this issue. 

From a quick look at the error trace, I think the issue is that the 1d `lon`/`lat` dimensions are not correctly renamed to `x`/`y` in the steps before `broadcast_lonlat` is applied. This seems to be something specific to this variable (or combination of variable and other parameters), since in my cloud tests ([here](https://github.com/jbusecke/cmip6_preprocessing/blob/9e0fffbd4e12e69ec19fd7e29077ce876212df21/tests/test_preprocessing_cloud.py#L170-L184)) it is not one of the `fail_cases`.

Let me fire up a notebook and get to the bottom of this.

**Comment by jbusecke** [comment]
Ok I think I found the culprit. I assume you are using a released version (0.1.4?). For me the above code works with the upstream master (installed with `pip install git+https://github.com/jbusecke/cmip6_preprocessing.git`). Could you try with that version and see if it works for you? I should release a new version soon to prevent this situation.

A more verbose explanation: I used to store model specific renaming information in a dictionary [from v0.1.4](https://github.com/jbusecke/cmip6_preprocessing/blob/ea8912333b0184484d7e5b758534e1f44cb8adfb/cmip6_preprocessing/preprocessing.py#L8-L104), but have since change that behavior to work for any model, to avoid having to upgrade this information any time a new model is added to the cloud store.

**Comment by dgergel** [comment]
@jbusecke that worked!! Thanks so much for testing this and figuring out what was going on. I'll go ahead and close this issue.
